app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -503,13 +504,27 @@ public void handleResult(@NonNull final List<PlaylistStreamEntry> result) {
         }
         setVideoCount(itemListAdapter.getItemsList().size());
 
-        playlistControlBinding.playlistCtrlPlayAllButton.setOnClickListener(view ->
-                NavigationHelper.playOnMainPlayer(activity, getPlayQueue()));
-        playlistControlBinding.playlistCtrlPlayPopupButton.setOnClickListener(view ->
-                NavigationHelper.playOnPopupPlayer(activity, getPlayQueue(), false));
-        playlistControlBinding.playlistCtrlPlayBgButton.setOnClickListener(view ->
-                NavigationHelper.playOnBackgroundPlayer(activity, getPlayQueue(), false));
-
+        playlistControlBinding.playlistCtrlPlayAllButton.setOnClickListener(view -> {
+            NavigationHelper.playOnMainPlayer(activity, getPlayQueue());
+            if (PreferenceManager.getDefaultSharedPreferences(activity)
+                    .getBoolean(getString(R.string.show_hold_to_append_key), true)) {
+                Toast.makeText(activity, R.string.hold_to_append, Toast.LENGTH_SHORT).show();
+            }
+        });
+        playlistControlBinding.playlistCtrlPlayPopupButton.setOnClickListener(view -> {
+            NavigationHelper.playOnPopupPlayer(activity, getPlayQueue(), false);
+            if (PreferenceManager.getDefaultSharedPreferences(activity)
+                    .getBoolean(getString(R.string.show_hold_to_append_key), true)) {
+                Toast.makeText(activity, R.string.hold_to_append, Toast.LENGTH_SHORT).show();
+            }
```

Comment:
Instead of duplicating these four lines of code three times, it's better if you create a function (e.g. right after the `handleResult`) named `showHoldToAppendTipIfNeeded` containing these four lines, and then you can just call it in the three places.

Reply:
Thanks for the feedback, I've put those four lines of code in a function